x-pack/plugins/spaces/server/plugin.ts

```diff
@@ -148,18 +148,28 @@ export class SpacesPlugin
       logger: this.log,
     });
 
-    const externalRouter = core.http.createRouter<SpacesRequestHandlerContext>();
-    initExternalSpacesApi({
-      externalRouter,
+    const buildFlavor = this.initializerContext.env.packageInfo.buildFlavor;
+    const router = core.http.createRouter<SpacesRequestHandlerContext>();
+    if (buildFlavor !== 'serverless') {
+      initPublicSpacesApi({
+        router,
+        log: this.log,
+        getStartServices: core.getStartServices,
+        getSpacesService,
+        usageStatsServicePromise,
+      });
+    }
+    const configuredAccess = buildFlavor === 'serverless' ? 'internal' : 'public';
+    initConfigurableSpacesApi({
```

Comment:
I differentiated the API's whose access depends on build flavor, but I left these in the "external" folder. It may make sense to move them into their own folder. I may also be beneficial to rename 'external' to 'public', so we would have 'public', 'internal', and 'configurable' APIs.

Reply:
As I mentioned in Slack, I'm a bit confused about these access tags, let's sync on that.

Reply:
@azasypkin I just pushed a commit to realign with expectations. Sorry for the confusion!